autoscheduler/scraper/management/commands/scrape_courses.py

```diff
@@ -8,128 +8,152 @@
+    for faculty_data in course_data['faculty']:
+        # We only care about the primary instructor, so skip all of the other ones
+        if not faculty_data['primaryIndicator']:
+            continue
+
+        name = faculty_data.get("displayName")
+
+        if name is None:
             return None
-        instructor_email = i['emailAddress']
-        #creates and saves instructor object
-        instructor_object = Instructor(
-            id=instructor_id, email_address=instructor_email)
-        instructor_object.save()
-        return instructor_object
+
+        email = faculty_data['emailAddress']
+
+        instructor_model = Instructor(id=name, email_address=email)
+        instructor_model.save()
+
+        return instructor_model
 
 def parse_course(course_data):
-    """puts course data in database. calls parse instructor and section"""
-    course_dept = course_data['subject']
+    """ Creates Course model and saves it to the databsae.
+        Calls parse_instructor and parse_section
+    """
+
+    dept = course_data['subject']
     course_number = course_data['courseNumber']
-    course_title = course_data['courseTitle']
-    course_term_code = course_data['term']
-    course_id = generate_course_id(
-        course_dept, course_number, course_term_code)#primary key
-    course_credit_hours = course_data['creditHourLow']
+    term_code = course_data['term']
+
+    # Generate the course's id using the above data
+    course_id = generate_course_id(dept, course_number, term_code)
+
+    title = course_data['courseTitle']
+    credit_hours = course_data['creditHourLow']
+
     #creates and saves course object
-    course_object = Course(
-        id=course_id, dept=course_dept, course_num=course_number, title=course_title,
-        credit_hours=course_credit_hours, term=course_term_code)
-    course_object.save()
-    #calls other parse functions
-    instructor_object = parse_instructor(course_data)
-    parse_section(course_data, instructor_object)
-
-def get_department_names(term_code) -> List[str]:
-    """queries database for list of all departments"""
+    course_model = Course(id=course_id, dept=dept, course_num=course_number,
+                          title=title, credit_hours=credit_hours, term=term_code)
+    course_model.save()
+
+    # Parse the instructor, then send the returned Instructor model to parse_section
+    instructor_model = parse_instructor(course_data)
+    parse_section(course_data, instructor_model)
+
+def get_department_names(term_code: str) -> List[str]:
+    """ Queries database for list of all departments """
     return [dept.code for dept in Department.objects.filter(term=term_code)]
 
 class Command(base.BaseCommand):
-    " Gets course information from banner and adds it to the database "
+    """ Gets course information from banner and adds it to the database """
 
     def add_arguments(self, parser):
         parser.add_argument('--term', type=str, required=False, default="201931")
```

Comment:
I'm not sure if you were planning on changing this to not require `--term` in this branch, but if so this should be changed